source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1758 @@
+
+  is-upper-boundᴮ : (x : ⟨ L ⟩)
+                  → (x is-an-upper-bound-of
+                       (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds
+  is-upper-boundᴮ x = pr₁ (is-supᴮ x)
+
+  is-least-upper-boundᴮ : (x : ⟨ L ⟩)
+                        → ((u' , _) : upper-bound
+                                      (small-↓ᴮ x , small-↓ᴮ-inclusion x))
+                        → (x ≤ u') holds
+  is-least-upper-boundᴮ x = pr₂ (is-supᴮ x)
+
+\end{code}
+
+Now we construct the least closed subset of an inductive definition as a QIT.
+Since HIT's are not native in Agda we postulate the existence of such a type
+as well as it's induction principle and work with it axiomatically. It is
+conveinient to package this postulate as a Record.
+
+Notice that the QIT has to constructors which representing the closure
+conditions we wish to impose on subsets. The c-cl constructor says that for
+any subset contained in the least closed subset elements in the downset of
+it's join are in the least closed subset. The ϕ-cl constructor says that for
+any a : L and b : B with (b , a) ∈ ϕ and ↓ᴮ a 'contained' in the least
+closed subset then b is in the least closed subset.
+
+We also derive the initiality of the least closed subset from the postulated
+induction principle. Initiality is closely related with the 'least' part of
+our least fixed point theorem.
+
+\begin{code}
+
+module Inductive-Definitions {𝓤 𝓦 𝓥 : Universe}
+                             {B : 𝓥  ̇}
+                             (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             (q : B → ⟨ L ⟩)
+                              where
+
+ open Small-Basis L q
+ open Joins _≤_
+
+ module Ind-from-Small-Basis-Facts (h : is-small-basis) where
+
+  open Small-Basis-Facts h
+
+  record Inductively-Generated-Subset-Exists (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)): 𝓤ω
+   where
+   constructor
+    inductively-generated-subset
+
+   field
+    Ind : B → 𝓤 ⊔ 𝓥 ⁺  ̇
+    Ind-trunc : (b : B) → is-prop (Ind b)
+    c-closed : (U : 𝓟 {𝓥} B)
+             → ((b : B) → (b ∈ U → Ind b))
+             → (b : B) → b ≤ᴮ (⋁ ((Σ b ꞉ B , b ∈ U) , q ∘ pr₁))
+             → Ind b
+    ϕ-closed : (a : ⟨ L ⟩)
+             → (b : B)
+             → (b , a) ∈ ϕ
+             → ((b' : B) → (b' ≤ᴮ a → Ind b'))
+             → Ind b
+    Ind-induction : (P : (b : B) → 𝓟 {𝓣} (Ind b))
+                  → ((U : 𝓟 {𝓥} B) → (f : (x : B) → (x ∈ U → Ind x))
+                   → ((x : B) → (u : x ∈ U) → f x u ∈ P x)
+                   → (b : B)
+                   → (g : (b ≤ᴮ (⋁ ((Σ x ꞉ B , U x holds) , q ∘ pr₁))))
+                   → c-closed U f b g ∈ P b)
+                  → ((a : ⟨ L ⟩)
+                   → (b : B)
+                   → (p : (b , a) ∈ ϕ)
+                   → (f : (x : B) → (x ≤ᴮ a → Ind x))
+                   → ((x : B) → (o : x ≤ᴮ a) → f x o ∈ P x)
+                   → ϕ-closed a b p f ∈ P b)
+                  → (b : B) → (i : Ind b) → i ∈ P b
+
+  module Trunc-Ind-Def (ϕ : 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩))
```

Comment:
Lowercase. (I will stop pointing this out. Please pay attention to case everywhere.)